src/sz.c

```diff
@@ -92,7 +92,8 @@ static void
 sz_boot_size2index_tab(const sc_data_t *sc_data) {
 	size_t dst_max = (SC_LOOKUP_MAXCLASS >> SC_LG_TINY_MIN) + 1;
 	size_t dst_ind = 0;
-	for (unsigned sc_ind = 0; sc_ind < SC_NSIZES && dst_ind < dst_max;
+	assert(SC_NSIZES <= 1 << (sizeof(uint8_t) * 8));
+	for (uint8_t sc_ind = 0; sc_ind < SC_NSIZES && dst_ind < dst_max;
```

Comment:
It looks sc_data_t has no explicit uint8 type. So the only uint8 part is `sz_size2index_tab`? If so let's add the assert and conversion to that specific part only.